samples/userspace/prod_consumer/src/sample_driver_foo.c

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright (c) 2019 Intel Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "sample_driver.h"
+#include <string.h>
+#include <kernel.h>
+#include <logging/log.h>
+
+LOG_MODULE_REGISTER(sample_driver);
+
+/*
+ * Fake sample driver for demonstration purposes
+ *
+ * This is fake driver for demonstration purposes, showing how an application
+ * can make system calls to interact with it.
```

Comment:
Just one minor comment, the syscall itself is not interacting with this driver, so this comment is a little bit confuse.

Reply:
the application is but I understood reading it that the syscall would communicate/interact with the driver